### PR TITLE
Remove bandalore src dependency

### LINKED ISSUE #30: Binary message attributes
Haven't looked into this much, but message attributes should check what the data type of the `MessageAttributeValue` is and call the correct getter.

```clojure
(defn- clojurify-message-attributes [^Message msg]
  (let [javafied-message-attributes (.getMessageAttributes msg)]
    (->> javafied-message-attributes
         (map (fn [[k ^MessageAttributeValue mav]] [(keyword k) (.getStringValue mav)]))
         (into {}))))
```

Untested but possible solution instead of `(.getStringValue mav)`:

```clojure
(case (.getDataType mav)
  "String" (.getStringValue mav)
  "Number" (try
             (BigDecimal. (.getStringValue mav))
             (catch NumberFormatException e
               {:error e :original-value (.getStringValue mav)}))
  "Binary" (.getBinaryValue mav))
```

### LINKED ISSUE #34: Create Queue Question
Hi.

I had a problem starting a project using this library on AWS which was caused by the fact the role on the EC2 instance didn't have the `sqs:CreateQueue` permission. The queues already existed and are managed with Terraform, so I don't want to give this permission out to the roles. I see at the moment this seems to be because it relies on the fact the Bandalore library provides `create-queue` which returns true if it already exists (but still required the CreateQueue permission).

What are your thoughts on this, would you be open to a PR checking the queues existence first?  Same for dead-letter queue.

Cheers

### PR DESCRIPTION
**Pull request by mtkp**
Directly use Java SQS API.

- use non-deprecated methods
- more efficient get and set queue attributes
- fix #34 (only create queue when needed)
- fix #30 (support binary message attributes)
- docstring/README polish

### CODE REVIEW

File: src/com/climate/squeedo/sqs.clj [hunk 1]
```diff
@@ -55,30 +62,39 @@
...
   Optionally takes a dead letter queue URL (Amazon Resource Name),
   a queue that already exists, that gets associated with the returned queue."
-    [client queue-name queue-atts & [dead-letter-url]]
-  (let [default-attrs {"DelaySeconds"                  0    ; delay after enqueue
+  [^AmazonSQSClient client ^String queue-name queue-attrs & [dead-letter-url]]
+  (let [default-attrs {"DelaySeconds"                  0       ; delay after enqueue
                        "MessageRetentionPeriod"        1209600 ; max, 14 days
                        "ReceiveMessageWaitTimeSeconds" poll-timeout-seconds
                        "VisibilityTimeout"             auto-retry-seconds}
         ; Tip: don't try sending attrs as create-queue's 'options' param: they aren't the same
```
Comment (tommy) [inline src/com/climate/squeedo/sqs.clj]: Nit: can remove this bandalore-specific comment.

File: src/com/climate/squeedo/sqs.clj [hunk 2]
```diff
@@ -55,30 +62,39 @@
...
+                            "deadLetterTargetArn" queue-arn})}))
 
 (defn- get-queue
   "Retrieve the queue URL (Amazon Resource Name) from SQS.
 
   Optionally takes a dead letter queue URL (Amazon Resource Name),
   a queue that already exists, that gets associated with the returned queue."
-    [client queue-name queue-atts & [dead-letter-url]]
-  (let [default-attrs {"DelaySeconds"                  0    ; delay after enqueue
+  [^AmazonSQSClient client ^String queue-name queue-attrs & [dead-letter-url]]
```
Comment (tommy) [inline src/com/climate/squeedo/sqs.clj]: Might change this varargs-based optional param to an arity-based one while you're here.

File: src/com/climate/squeedo/sqs.clj [hunk 3]
```diff
@@ -55,30 +62,39 @@
...
+                    (.getQueueUrl (.getQueueUrl client queue-name))
+                    (catch QueueDoesNotExistException _
+                      (.getQueueUrl (.createQueue client queue-name))))
+        attributes (->> (merge default-attrs
+                               queue-attrs
+                               (when dead-letter-url
+                                 (redrive-policy client dead-letter-url)))
+                        (map (fn [[k v]] [(str k) (str v)]))
+                        (into {}))]
+    (.setQueueAttributes client queue-url attributes)
```
Comment (tommy) [inline src/com/climate/squeedo/sqs.clj]: Doing this after the createQueue/getQueueUrl operation requires the SetQueueAttributes permission (even if the queue exists and the attributes are already set as expected, and even if no DLQ is specified).

I see three options:
1) current behavior (always try to set attrs for a queue even if it exists)
2) only set attrs when creating the queue (if a queue exists with attrs that don't match, do not update them)
3) identify if existing attrs don't match expected ones and update appropriately.

My recommendation is #2 (and maybe issuing a warn if DLQ or queue-attrs are specified, but the queue exists already?). Reason being: a) don't want to require SetQueueAttributes permission in order to use squeedo as a consumer and b) squeedo probably shouldn't be a queue attribute diffing+migration tool.

Open to other thoughts.

Comment (rodnaph) [inline src/com/climate/squeedo/sqs.clj]: My 2 cents...  As a user of squeedo (unless there's a compelling reason I'm missing...) I'd prefer it didn't try to manage queues and their configuration for me, that's already done by my configuration management (Terraform).

File: src/com/climate/squeedo/sqs.clj (hunk 1 above)
Comment (mtkp) [inline src/com/climate/squeedo/sqs.clj]: thanks, removed

File: src/com/climate/squeedo/sqs.clj (hunk 2 above)
Comment (mtkp) [inline src/com/climate/squeedo/sqs.clj]: good idea, done

File: src/com/climate/squeedo/sqs.clj (hunk 3 above)
Comment (mtkp) [inline src/com/climate/squeedo/sqs.clj]: thanks guys. i'm going with option 2, it seems reasonable. i've added a new function, `set-queue-attributes` to provide a way for users to opt-in to queue attribute updates if they need it.

File: src/com/climate/squeedo/sqs.clj [hunk 4]
```diff
@@ -90,33 +124,50 @@
...
+
+  Note: if the queue already exists, the queue attributes and redrive policy
+  will not be modified. Use `set-queue-attributes` to set attributes on an
+  existing queue."
   [queue-name & {:keys [dead-letter
                         client
                         queue-attributes
                         dead-letter-queue-attributes]}]
   (validate-queue-name! queue-name)
-  (when dead-letter
```
Comment (mtkp) [inline src/com/climate/squeedo/sqs.clj]: this is redundant thanks to the dead-letter recursion on mk-connection

**Pull request merged**